.github/workflows/go-ci.yaml

```diff
@@ -14,75 +14,82 @@ jobs:
-        ref: ${{ needs.get_refs.outputs.tenant-security-proxy }}
-        path: tenant-security-proxy
-        token: ${{ secrets.WORKFLOW_PAT }}
-    - uses: actions/cache@v2
-      with:
-        # In order:
-        # * Module download cache
-        # * Build cache (Linux)
-        # * Build cache (Mac)
-        # * Build cache (Windows)
-        # * Cargo registry
-        # * Cargo packages
-        # * TSP build
-        path: |
-          ~/go/pkg/mod
-          ~/.cache/go-build
-          ~/Library/Caches/go-build
-          ~\AppData\Local\go-build
-          ~/.cargo/registry
-          ~/.cargo/git
-          tenant-security-proxy/target
-        key: ${{ runner.os }}-go-${{ matrix.go-version }}-${{ hashFiles('**/go.sum') }}
-        restore-keys: |
-          ${{ runner.os }}-go-${{ matrix.go-version }}-
-    - name: Decrypt TSC integration keys
-      uses: IronCoreLabs/ironhide-actions/decrypt@v2
-      with:
-        keys: ${{ secrets.IRONHIDE_KEYS }}
-        input: .env.integration.iron
-    - name: install zmq (linux)
-      if: ${{ runner.os == 'Linux' }}
-      run: sudo apt update && sudo apt install -y --no-install-recommends libzmq3-dev
-    - name: install zmq (mac)
-      if: ${{ runner.os == 'macOS' }}
-      run: brew install zmq coreutils
-    - name: Build TSP
-      working-directory: tenant-security-proxy
-      run: cargo build --release
-    - name: Run tests
-      run: |
-        cd tenant-security-proxy && env $(cat ../.env.integration) cargo run --release &
-        timeout 700 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:9000/ready)" =~ ''[01346-9][0-9][0-9]'' ]]; do sleep 5; done' || false
-        env $(cat .env.integration) go test ./...
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - uses: actions/checkout@v3
+      - name: clone the tsp
+        uses: actions/checkout@v3
+        with:
+          repository: IronCoreLabs/tenant-security-proxy
+          ref: ${{ needs.get_refs.outputs.tenant-security-proxy }}
+          path: tenant-security-proxy
+          token: ${{ secrets.WORKFLOW_PAT }}
+      - uses: actions/cache@v3
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+          # * Cargo registry
+          # * Cargo packages
+          # * TSP build
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/Library/Caches/go-build
+            ~\AppData\Local\go-build
+            ~/.cargo/registry
+            ~/.cargo/git
+            tenant-security-proxy/target
+          key: ${{ runner.os }}-go-${{ matrix.go-version }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-${{ matrix.go-version }}-
+      - name: Decrypt TSC integration keys
+        uses: IronCoreLabs/ironhide-actions/decrypt@v3
```

Comment:
I think I'll add Dependabot for the GitHub actions like you suggested, but I'm just going to turn on Security Updates for the Go packages instead of all updates.